src/main/java/org/commcare/suite/model/Profile.java

```diff
@@ -208,6 +215,7 @@ public void readExternal(DataInputStream in, PrototypeFactory pf)
         properties = (Vector<PropertySetter>)ExtUtil.read(in, new ExtWrapList(PropertySetter.class), pf);
         roots = (Vector<RootTranslator>)ExtUtil.read(in, new ExtWrapList(RootTranslator.class), pf);
         featureStatus = (Hashtable<String, Boolean>)ExtUtil.read(in, new ExtWrapMap(String.class, Boolean.class), pf);
+        buildProfileId = ExtUtil.readString(in);
```

Comment:
@ctsims  I didn't find any particular code that serializes Profile so I think we don't need to worry about any sort of migration from old Profile to new Profile. Let me know if you think that's not right.

Reply:
I think that makes sense on commcare-android, since we read/write the profile from disk as the raw file. On J2ME I believe we stored the serialized version instead.

Can you check the `formplayer` process, though? I was surprised to see that a number of formplayer resources which I figured were stored and read as xml files were instead using the serialized versions. If so, all we need to do is bump the application database version on formplayer to force a full reinstall.

Reply:
confirming that this will require a formplayer app version dump.